Mathlib/Order/Filter/Basic.lean

```diff
@@ -3325,3 +3325,23 @@ theorem Set.MapsTo.tendsto {α β} {s : Set α} {t : Set β} {f : α → β} (h
     Filter.Tendsto f (𝓟 s) (𝓟 t) :=
   Filter.tendsto_principal_principal.2 h
 #align set.maps_to.tendsto Set.MapsTo.tendsto
+
+/-- Construct a filter from a property that is stable under finite intersections.
+This constructor deduces `Filter.univ_sets` and `Filter.inter_sets` from the finite
+intersection property. -/
+def Filter.ofProp_finite_inter {p : Set α → Prop} (he : p univ) (hmono : ∀ ⦃s t⦄, s ⊆ t → p s → p t)
+    (hinter : ∀ ⦃s t⦄, p s → p t → p (s ∩ t)) : Filter α where
+  sets := {t | p t}
+  univ_sets := by simpa
+  sets_of_superset := fun {s t} a a_1 ↦ hmono a_1 a
+  inter_sets := fun {s t} hs ht ↦ hinter hs ht
+
+/-- Construct a filter from a property that is stable under finite unions.
+This constructor deduces `Filter.univ_sets` and `Filter.inter_sets` from the finite
+intersection property. -/
+def Filter.ofProp_finite_union {p : Set α → Prop} (he : p ∅) (hmono : ∀ ⦃s t⦄, s ⊆ t → p t → p s)
+    (hunion : ∀ ⦃s t⦄, p s → p t → p (s ∪ t)) : Filter α where
```

Comment:
This can be called `comk` similarly to [Filter.coext](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Order/Filter/Basic.html#Filter.coext).

Reply:
I've renamed it, but I don't think modifying 'def cofinite' would be useful, as (he : p \empty) fails for {s : s\compl is finite}. That would require the other def Filter.ofProp_finite_inter from the previous comment. I haven't added the new constructor yet for cofinite, let me know if you want me to keep Filter.ofProp_finite_inter in!

Reply:
It's `p = Set.Finite`, not `p s = Set.Finite sᶜ`. Fixed. BTW, could you please use backticks for code in your comments?